### ISSUE TITLE
Fix error reporting

### ISSUE THREAD

**Issue by youriwims** [mozilla/donate-wagtail#343]
### Issue: Insufficient error reporting

**Element path:** _multiple instances_

**Snippet:** _multiple instances_

**How to fix:**
---
[key] 👇 
 - Issue
    - How to Fix
---
**Error Identification**
- Donate home 'Other Amount' `<input>` has no error state for empty, invalid inputs.
   - Add error states/logic

**Error Suggestion**
- Email input on cc payment form presents no error as long as '@' is included
   - Add error state/logic
- Zip code has no error for random, long input
   - Add error state/logic
- Expiration data allows either 2 or 4 digit year
   - Add helper/placeholder text that displays proposed input format
  - Add logic to only accept one of the formats

**Error Prevention**
- There's no _Review_, _Confirm_, or _Correct_ flow after submitting payment data
   - ???

**Status Messages**
- The amount box at the top of the payment form should have a `role` of `log` or `status`
   - All form notifications(mostly at the top in gray boxes) that align with the list below should have a `role` of `log`, `alert`, or `status`:
     - The success or results of an action
     - The waiting state of an application
     - The progress of a process
     - The existence of errors

====

More Info:

Comments: Many of these accessibility concerns overlap with ux/ui...may be good to get @beccaklam and @alanmoo input.
Resources: 
- Related to #305 
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_log_role
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_status_role
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alert_role

**Comment by philwakefield** [comment]
@alanmoo Not sure the exact requirements to resolve this ticket. This may also be several issues in 1 ticket. Please can you provide acceptance criteria to resolve this ticket.

**Comment by youriwims** [comment]
@philwakefield would it be better to break this one up? Let me know and I can separate the tickets.

**Comment by youriwims** [comment]
I highlighted the items of most importance from this ticket in: #363, #366 & #367. They're also included in master a11y qa ticket checklist, #257.

**Comment by Pomax** [comment]
closing as out of date
